iDiff/differs/pipDiff.go

```diff
@@ -0,0 +1,117 @@
+package differs
+
+import (
+	"io/ioutil"
+	"path/filepath"
+	"regexp"
+	"strconv"
+
+	"github.com/GoogleCloudPlatform/runtimes-common/iDiff/utils"
+	"github.com/golang/glog"
+)
+
+// layers of two different images.
+func PipDiff(d1file, d2file string) (string, error) {
+	d1, err := utils.GetDirectory(d1file)
+	if err != nil {
+		glog.Errorf("Error reading directory structure from file %s: %s\n", d1file, err)
+		return "", err
+	}
+	d2, err := utils.GetDirectory(d2file)
+	if err != nil {
+		glog.Errorf("Error reading directory structure from file %s: %s\n", d2file, err)
+		return "", err
+	}
+
+	dirPath1 := d1.Root
+	dirPath2 := d2.Root
+	pack1 := getPythonPackages(dirPath1)
+
+	pack2 := getPythonPackages(dirPath2)
+
+	diff := utils.DiffMaps(pack1, pack2)
+	diff.Image1 = dirPath1
+	diff.Image2 = dirPath2
+	output(diff)
+	return "", nil
+}
+
+func getPythonVersion(pathToLayer string) (string, bool) {
+	libPath := filepath.Join(pathToLayer, "/layer/usr/local/lib")
+	libContents, err := ioutil.ReadDir(libPath)
+	if err != nil {
+		return "", false
+	}
+
+	for _, file := range libContents {
+		pattern := regexp.MustCompile("^python[0-9]+\\.[0-9]+$")
+		match := pattern.FindString(file.Name())
+		if match != "" {
+			return match, true
+		}
+	}
+	return "", false
+}
+
+// TODO: Eventually, this would make use of the shallow JSON and be diffed
+// with that of another image to get only the layers that have changed.
+func getImageLayers(pathToImage string) []string {
```

Comment:
I think you could similarly do this with the BuildLayerTarget function by passing an empty string for the target.